lib/optimization/table_data.go

```diff
@@ -131,8 +131,13 @@ func (t *TableData) InsertRow(pk string, rowData map[string]any, delete bool) {
 	if prevRow, isOk := t.rowsData[pk]; isOk {
 		prevRowSize = size.GetApproxSize(prevRow)
 		if delete {
-			// If the row was deleted, preserve the previous values that we have in memory
+			// If the row was deleted, preserve the previous values that we have in memory,
+			// unless the previous row was also a delete operation
 			rowData = prevRow
+			prevRowDeleted, isOk := prevRow[constants.DeleteColumnMarker]
+			if !isOk || !prevRowDeleted.(bool) {
+				rowData[constants.OnlySetDeleteColumnMarker] = false
```

Comment:
This is where we override the marker to set it to false if realize we actually have the previous values in memory, meaning we want to update all columns (not just the `__artie_delete` column)